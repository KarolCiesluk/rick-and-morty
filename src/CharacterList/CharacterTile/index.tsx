interface CharacterTileProps {
  image: string;
  name: string;
  status: string;
  species: string;
}

const CharacterTile = ({ image, name, status, species }: CharacterTileProps) => (
  <article>
    <img src={image} alt="" />
    <h3>{name}</h3>
    <p>Status: {status}</p>
    <p>Species: {species}</p>
  </article>
);

export default CharacterTile;