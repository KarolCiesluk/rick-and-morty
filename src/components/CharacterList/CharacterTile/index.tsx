import { Image, CharacterInfo } from "./styled";

interface CharacterTileProps {
  image: string;
  name: string;
  status: string;
  species: string;
}

const CharacterTile = ({ image, name, status, species }: CharacterTileProps) => (
  <article>
    <Image src={image} alt="" />
    <h3>{name}</h3>
    <CharacterInfo data-testid="status">
      Status: <b>{status}</b>
    </CharacterInfo>
    <CharacterInfo data-testid="species">
      Species: <b>{species}</b>
    </CharacterInfo>
  </article>
);

export default CharacterTile;