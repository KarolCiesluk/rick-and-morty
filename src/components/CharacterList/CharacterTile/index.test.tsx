import renderer from "react-test-renderer";

import CharacterTile from ".";

describe("CharacterTile tests", () => {
  const exampleCharacter = {
    "name": "Crocubot",
    "status": "Dead",
    "species": "Animal",
    "image": "https://rickandmortyapi.com/api/character/avatar/81.jpeg",
  };

  it("renders ok", () => {
    const tree = renderer
      .create(
        <CharacterTile
          name={exampleCharacter.name}
          status={exampleCharacter.status}
          species={exampleCharacter.species}
          image={exampleCharacter.image}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});