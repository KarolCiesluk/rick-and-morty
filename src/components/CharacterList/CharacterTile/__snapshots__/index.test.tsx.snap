// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterTile tests renders ok 1`] = `
<article>
  <img
    alt=""
    className="sc-bdvvtL eukWAO"
    src="https://rickandmortyapi.com/api/character/avatar/81.jpeg"
  />
  <h3>
    Crocubot
  </h3>
  <p
    className="sc-gsDKAQ fiIEyU"
  >
    Status: 
    <b>
      Dead
    </b>
  </p>
  <p
    className="sc-gsDKAQ fiIEyU"
  >
    Species: 
    <b>
      Animal
    </b>
  </p>
</article>
`;
